arches/app/models/resource.py

```diff
@@ -522,31 +528,28 @@ def get_related_resources(
         ret = {"resource_instance": self, "resource_relationships": [], "related_resources": [], "node_config_lookup": graph_lookup}
 
         if page > 0:
-            limit = settings.RELATED_RESOURCES_PER_PAGE
-            start = limit * int(page - 1)
+            number_per_page = settings.RELATED_RESOURCES_PER_PAGE
+            start = number_per_page * int(page - 1)
+            limit = number_per_page * page
 
         def get_relations(resourceinstanceid, start, limit, resourceinstance_graphid=None):
-            query = Query(se, start=start, limit=limit)
-            bool_filter = Bool()
-            bool_filter.should(Terms(field="resourceinstanceidfrom", terms=resourceinstanceid))
-            bool_filter.should(Terms(field="resourceinstanceidto", terms=resourceinstanceid))
+            final_query = Q(resourceinstanceidfrom_id=resourceinstanceid) | Q(resourceinstanceidto_id=resourceinstanceid)
 
             if resourceinstance_graphid:
-                graph_filter = Bool()
-                to_graph_id_filter = Bool()
-                to_graph_id_filter.filter(Terms(field="resourceinstancefrom_graphid", terms=str(self.graph_id)))
-                to_graph_id_filter.filter(Terms(field="resourceinstanceto_graphid", terms=resourceinstance_graphid))
-                graph_filter.should(to_graph_id_filter)
-
-                from_graph_id_filter = Bool()
-                from_graph_id_filter.filter(Terms(field="resourceinstancefrom_graphid", terms=resourceinstance_graphid))
-                from_graph_id_filter.filter(Terms(field="resourceinstanceto_graphid", terms=str(self.graph_id)))
-                graph_filter.should(from_graph_id_filter)
-                bool_filter.must(graph_filter)
+                to_graph_id_filter = Q(resourceinstancefrom_graphid_id=str(self.graph_id)) & Q(
+                    resourceinstanceto_graphid_id=resourceinstance_graphid
+                )
+                from_graph_id_filter = Q(resourceinstancefrom_graphid_id=resourceinstance_graphid) & Q(
+                    resourceinstanceto_graphid_id=str(self.graph_id)
+                )
+                final_query = final_query & (to_graph_id_filter | from_graph_id_filter)
 
-            query.add_query(bool_filter)
+            relations = {
+                "total": models.ResourceXResource.objects.filter(final_query).count(),
+                "relations": models.ResourceXResource.objects.filter(final_query)[start:limit],
+            }
 
-            return query.search(index=RESOURCE_RELATIONS_INDEX)
+            return relations  # resourceinstance_graphid = "00000000-886a-374a-94a5-984f10715e3a"
```

Comment:
Could not open the report page and found this function has changed from using Elasticsearch to the database so @RobOatesHistoricEngland code on ln:560-574 needs to be changed to account for this.

Reply:
@aj-he - Please retest